cdci_data_analysis/analysis/instrument.py

```diff
@@ -684,6 +700,17 @@ def set_input_products_from_fronted(self, input_file_path, par_dic, verbose=Fals
             else:
                 raise RuntimeError
 
+    def update_par_dic_with_uploaded_files(self, par_dic, uploaded_files_obj, products_url):
+        if validators.url(products_url):
+            basepath = os.path.join(products_url, 'dispatch-data/download_file')
+        else:
+            basepath = os.path.join(products_url, 'download_file')
```

Comment:
why is this needed?

Reply:
Simply in case the products_url is a string during tests (ie `PRODUCTS_URL`) 

Same approach has been used for the `instr-list` endpoint